core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -165,6 +168,7 @@ public static void registerPrototype(String type, Custom proto) {
     private final ImmutableOpenMap<String, IndexMetaData> indices;
     private final ImmutableOpenMap<String, IndexTemplateMetaData> templates;
     private final ImmutableOpenMap<String, Custom> customs;
+    private final ImmutableOpenMap<String, IndexTombstone> indexTombstones;
```

Comment:
do they need to be keyed by the index name? I think it can just be a List of tombstones?